components/esp_websocket_client/examples/linux/CMakeLists.txt

```diff
@@ -6,10 +6,12 @@ set(common_component_dir ../../../../common_components)
 set(EXTRA_COMPONENT_DIRS
    ../..
   "${common_component_dir}/linux_compat/esp_timer"
-  "${common_component_dir}/linux_compat"
   "${common_component_dir}/linux_compat/freertos"
    $ENV{IDF_PATH}/examples/protocols/linux_stubs/esp_stubs
    $ENV{IDF_PATH}/examples/common_components/protocol_examples_common)
 
 set(COMPONENTS main)
 project(websocket)
+
+idf_component_get_property(websocket esp_websocket_client COMPONENT_LIB)
+target_compile_definitions(${websocket} PRIVATE "-DCONFIG_IDF_TARGET_LINUX")
```

Comment:
wouldn't it be easier to `#include "sdkconfig.h"` in the relevant modules
(if they uses those `CONFIG_...` defines, they should include it)